contractcourt/commit_sweep_resolver.go

```diff
@@ -252,18 +257,26 @@ func (c *commitSweepResolver) Resolve(_ bool) (ContractResolver, error) {
 
 		signDesc = c.commitResolution.SelfOutputSignDesc
 	)
+
 	switch {
 	// For taproot channels, we'll know if this is the local commit based
-	// on the witness script. For local channels, the witness script has an
-	// OP_DROP value.
-	//
-	// TODO(roasbeef): revisit this after the script changes
-	//  * otherwise need to base off the key in script or the CSV value
-	//  (script num encode)
+	// on the timelock value. For remote commitment transactions, the
+	// witness script has a timelock of 1.
 	case c.chanType.IsTaproot():
-		scriptLen := len(signDesc.WitnessScript)
-		isLocalCommitTx = signDesc.WitnessScript[scriptLen-1] ==
-			txscript.OP_DROP
+		delayKey := *c.localChanCfg.DelayBasePoint.PubKey
+		nonDelayKey := *c.localChanCfg.PaymentBasePoint.PubKey
```

Comment:
looks like `nonDelayKey` provides little utility but sanity checking an impossible case?

Reply:
If we allow ourselves to assume that the signing key MUST be one of these two keys, then all we need is for one of them to be there as we can infer that if it isn't one, then it is the other. I chose this approach for thoroughness but we can trim it down if you'd rather this either/or assumption to be documented in the comments rather than explicitly checked in the control logic.